v2/datastream-to-spanner/terraform/samples/all-infrastructure/README.md

```diff
@@ -0,0 +1,554 @@
+## Sample Scenario: MySQL to Spanner using MySQL source configuration
+
+> **_SCENARIO:_** This Terraform example illustrates launching a live migration
+> job for a MySQL source, setting up all the required cloud infrastructure.
+> **Only the source details are needed as input.**
+
+## Terraform permissions
+
+In order to create the resources in this sample,
+the`Service account`/`User account` being used to run Terraform
+should have the
+required [permissions](https://cloud.google.com/iam/docs/manage-access-service-accounts#multiple-roles-console).
+There are two ways to add permissions -
+
+1. Adding pre-defined roles to the service account running Terraform.
+2. Creating a custom role with the granular permissions and attaching it to the
```

Comment:
is there a way to verify if these permissions exist before folks start ?
Can we add it in the terraform init/plan ?

Reply:
This can't be done with terraform init/plan.
I tried without permissions it fails in 5 seconds and the error is pretty intuitive. TF complains of the specific permissions that are missing.